AsyncDisplayKit/ASTextNode.mm

```diff
@@ -299,6 +287,26 @@ - (void)_invalidateRenderer
   _renderer = nil;
 }
 
+- (BOOL)_needInvalidateRenderer:(CGSize)newSize
+{
+  return !CGSizeEqualToSize(newSize, _constrainedSize);
+}
+
+- (void)_invalidateRendererIfNeeded {
+  [self _invalidateRendererIfNeeded:self.view.bounds.size];
```

Comment:
This takes makes the assumption that the node will always be view backed. Should take into account layer-backed text nodes.